### PR TITLE
Update versions list page to use publishing-dataset-controller

### PR DESCRIPTION
**Pull request by jondewijones**
### What

Update versions list page to use publishing-dataset-controller

### Who can review

Anyone

### CODE REVIEW

File: src/app/views/datasets-new/versions/DatasetVersionsController.jsx [hunk 1]
```diff
@@ -39,27 +36,28 @@ export class DatasetVersionsController extends Component {
...
-    getDataset = datasetID => {
-        this.setState({ isFetchingDataset: true });
+    getAllVersions = (datasetID, editions) => {
+        this.setState({ isFetchingData: true });
         return datasets
-            .get(datasetID)
-            .then(dataset => {
+            .getVersionsList(datasetID, editions)
+            .then(response => {
+                console.log(response);
```
Comment (rav-pradhan) [inline src/app/views/datasets-new/versions/DatasetVersionsController.jsx:46]: We should remove this log

**Review by rav-pradhan — COMMENTED** [review]
Looks good, tested locally as well and seemed to run as before. Just a couple tiny comments

File: src/app/views/datasets-new/versions/DatasetVersionsController.jsx (hunk 1 above)
Comment (jondewijones) [reply src/app/views/datasets-new/versions/DatasetVersionsController.jsx:46]: We should and have :)

File: src/app/utilities/api-clients/datasets.js [hunk 2]
```diff
@@ -210,4 +210,10 @@ export default class datasets {
             return response;
         });
     }
+
+    static getVersionsList(datasetID, editionID) {
+        return http.get(`${datasetControllerURL}/datasets/${datasetID}/editions/${editionID}/versions`).then(response => {
```
[the remark this answers is not in the document]
Comment (jondewijones) [reply src/app/utilities/api-clients/datasets.js:215]: Agreed. Clearer is always better. I've changed the variable name in this file for now. I think to change it through the codebase would be a separate task

**Review by rav-pradhan — APPROVED** [review]

**Pull request merged**